harmonia/src/narinfo.rs

```diff
@@ -113,10 +113,14 @@ fn query_narinfo(
     for sk in sign_keys {
         let fingerprint = fingerprint_path(store_path, &res.nar_hash, res.nar_size, &refs)?;
         if let Some(fp) = fingerprint {
-            res.sig.push(libnixstore::sign_string(sk, &fp)?);
+            res.sigs.push(libnixstore::sign_string(sk, &fp)?);
         }
     }
 
+    if res.sigs.is_empty() {
```

Comment:
I am now wondering if we not just return a 404 if some path is not signed or have a flag for that? Otherwise I am not sure how having signatures in some cases but not in other would be useful to have, because one might often get warnings in that case.

Reply:
I'm not sure whether returning 404 would be useful. It's up to the client to decide whether unsigned paths are OK or not.

I'm using this on a server which is shared by several people and runs CI for a few projects. Each project CI has its own signing key and I want the users of the project to only fetch binaries signed by the project CI. We see the warnings from time to time, but it's IMO OK.

`nix-serve` behaves the same as in this PR and I don't see a problem with that. But maybe, I'm missing something.

Reply:
If it's not a fatal error, it's ok, I guess.